src/MoBi.UI/Views/BasePathAndValueEntityView.cs

```diff
@@ -232,10 +234,28 @@ private void initColumnVisibility()
       {
          for (var i = 0; i < _pathElementsColumns.Count; i++)
          {
-            _pathElementsColumns[i].Visible = _presenter.HasAtLeastOneValue(i);
+            var shouldShow = _presenter.HasAtLeastOneValue(i);
+            var appearing = !_pathElementsColumns[i].Visible && shouldShow;
+            _pathElementsColumns[i].Visible = shouldShow;
+
+            if (appearing)
+               initializePathColumnVisibleIndex(i);
          }
       }
 
+      // When hiding and showing columns automatically based on presence of path elements
+      // we have to override how the visibility index is set. That's because visibility index
+      // of a column is updated as other columns are hidden. So, if you iterate in order and hide all columns
+      // then their internal hidden visibility index will be 0. If you iterate in order and show them,
+      // they appear in the reverse order
+      private void initializePathColumnVisibleIndex(int i)
```

Comment:
I found that when an empty parameter values editor suddenly gets parameter values the column order is messed up. It's the same with Initial Conditions.

If you closed and opened the editor, the path column order would be fixed.